frontend/cypress/integration/basic-spec.ts

```diff
@@ -0,0 +1,84 @@
+import { BigNumber } from 'ethers';
+import { expect } from 'chai';
+
+describe('Collateral auctions', function () {
+    // Set the private key to one of the deterministic hardhat keys.
+    // see https://hardhat.org/hardhat-network#running-stand-alone-in-order-to-support-wallets-and-other-software
+    const KEEPER_WALLET_PRIVATE_KEY = '0x701b615bbdfb9de65240bc28bd21bbc0d996645a3dd57e7b12bc2bdf6f192c82';
+    const URL_TO_VISIT = `${process.env.FRONTEND_URL || 'localhost:3000'}/collateral?network=localhost`;
+    let daiBalanceBefore: Record<string, BigNumber>;
+    let daiBalanceAfter: Record<string, BigNumber>;
+
+    function testSwapProfit() {
+        cy.visit(URL_TO_VISIT);
+        cy.window()
+            .its('$nuxt')
+            .then(function (nuxt) {
+                nuxt.$store.dispatch('wallet/createWalletFromPrivateKey', KEEPER_WALLET_PRIVATE_KEY);
+            });
+        cy.get('table').first({ timeout: 140000 }).should('contain.text', 'ETH');
+
+        cy.window()
+            .its('$nuxt')
+            .then(function (nuxt) {
+                daiBalanceBefore = nuxt.$store.getters['wallet/walletBalances'];
+            });
+
+        cy.get('a').eq(1).should('contain.text', 'Participate').click();
+        cy.get('button.Button').eq(3).should('contain.text', 'Directly swap').click();
+        cy.get('button.Button')
+            .eq(5)
+            .should('contain.text', 'Authorize DAI Transactions')
+            .click()
+            .then(() =>
+                cy
```

Comment:
Why do you sometimes need to put another chain of executions inside then? Can't it be moved to the new line, since cypress still magically "awaits" every line